### DISCUSSION TITLE
SMTP Authentication...

### DISCUSSION THREAD

**Discussion by martinpruss** [elsa-workflows/elsa-core#245] [Q&A] [answered]
Hi,

I've searched extensively but could not locate how to pass credentials for SMTP authentication.

Can somebody help please?

Thanks!

**Comment by sfmskywalker** [comment, marked as answer]
If you use `appsettings.json`, you would configure it like this:

```json
{
   "Smtp": {
        "Host": "smtp.sendgrid.net",
        "Port": "25",
        "Username": "some-user-name",
        "Password": "some-pwd-123"
    }
}
```

In `Startup`, you would configure Elsa + support for the `SendEmail` activity like this:

```csharp
public void ConfigureServices(IServiceCollection services)
{
    services
        .AddElsa()
        .AddEmailActivities(options => options.Bind(Configuration.GetSection("Smtp")));
}
```

Or you could do it hardcoded, like so:

```csharp
public void ConfigureServices(IServiceCollection services)
{
    services
        .AddElsa()
        .AddEmailActivities(options => options.Configure(smtpOptions =>
        {
            smtpOptions.Host = "smtp.sendgrid.net";
            smtpOptions.Port = 25;
            smtpOptions.UserName = "some-user-name";
            smtpOptions.Password = "some-pwd-123";
        }));
}
```

Hope this helps.

**Comment by sfmskywalker** [comment]
PS. See https://github.com/elsa-workflows/elsa-core/blob/develop/src/activities/Elsa.Activities.Email/Options/SmtpOptions.cs for all of the available options.

**Reply by MaddyDecoder** [reply]
its showing page not fpund i am also getting error of access denied

**Comment by martinpruss** [comment]
Awesome!  Thanks for the fast response, I really appreciate it.
